ppr-ui/src/composables/mhrRegistration/useNewMhrRegistration.ts

```diff
@@ -0,0 +1,120 @@
+import { useGetters } from 'vuex-composition-helpers'
+import {
+  getMhrAttentionReferenceNum,
+  getMhrRegistrationDocumentId,
+  getMhrRegistrationHomeDescription, getMhrRegistrationHomeOwnerGroups, getMhrRegistrationLocation,
+  getMhrRegistrationSubmittingParty
+} from '@/store/getters'
+import {
+  MhrRegistrationDescriptionIF,
+  MhrRegistrationHomeLocationIF,
+  MhrRegistrationHomeOwnerGroupIF,
+  NewMhrRegistrationApiIF
+} from '@/interfaces'
+
+export const useNewMhrRegistration = () => {
+  const {
+    getMhrRegistrationHomeDescription,
+    getMhrRegistrationSubmittingParty,
+    getMhrRegistrationDocumentId,
+    getMhrAttentionReferenceNum,
+    getMhrRegistrationLocation,
+    getMhrRegistrationHomeOwnerGroups
+  } = useGetters<any>([
+    'getMhrRegistrationHomeDescription',
+    'getMhrRegistrationSubmittingParty',
+    'getMhrRegistrationDocumentId',
+    'getMhrAttentionReferenceNum',
+    'getMhrRegistrationLocation',
+    'getMhrRegistrationHomeOwnerGroups'
+  ])
+
+  function cleanEmpty<Type> (obj:Type): Type {
+    const newObj = {}
+    Object.keys(obj).forEach((key) => {
+      if (obj[key] !== null && typeof obj[key] === 'object') newObj[key] = cleanEmpty(obj[key])
+      else if (!!obj[key] || obj[key] === 0) newObj[key] = obj[key]
+    })
+    return newObj as Type
+  }
+
+  const parseSubmittingParty = () => {
+    const submittingParty = getMhrRegistrationSubmittingParty.value
+
+    if (submittingParty.businessName) {
+      delete submittingParty.personName
+    } else {
+      delete submittingParty.businessName
+      if (!submittingParty.personName.middle) {
+        delete submittingParty.personName.middle
+      }
+    }
+    if (!submittingParty.emailAddress) {
+      delete submittingParty.emailAddress
+    }
+    if (!submittingParty.phoneNumber) {
+      delete submittingParty.phoneNumber
+    }
+    if (!submittingParty.phoneExtension) {
+      delete submittingParty.phoneExtension
+    }
+
+    return submittingParty
```

Comment:
To be transparent, this isn't how i would have done it. ie removing unwanted props vs constructing something new out of what WAS wanted. 
But also, i don't mind it at all, the only risk i can think of is unwanted properties in the payload. 
ie a developer adds a new property to store or something similar that doesn't have any handling here. 

I suppose it goes both ways since it would be included in the payload without additional work but it could also be included unintentionally. 

Either way, i'm fine with this! Just my thoughts no action required.

Reply:
hmm I think you're making more sense and I think will have less code, I'll look into it

Reply:
No action required from me but if you feel there is a better way by all means.
100% your call.